File: me_daily/2025/11/01/106073/cfgym106073.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

	int n, q;
	cin >> n >> q;

	vector<int> nums(q);
	for (auto &v: nums) cin >> v;

	int mod = 1e9 + 7, rev2 = (mod + 1) / 2;
	vector<int> ans(n + 1, 0);

	int cur = 0;
	reverse(nums.begin(), nums.end());

	for (auto &v: nums) {
		ans[v] = (1ll * cur * rev2 + ans[v]) % mod;
		cur = 1ll * (cur + v) * rev2 % mod;
	}

	ans[1] = (ans[1] + cur) % mod;

	for (int i = 1; i <= n; i ++)
		cout << ans[i] << '\n';

	return 0;
}

File: me_daily/2025/11/01/106073/cfgym106073.py
from collections import defaultdict
from typing import List
import sys
from collections import Counter
input = lambda: sys.stdin.readline().strip()
II = lambda: int(input())
LII = lambda: list(map(int, input().split()))
SI = lambda: input()
LSI = lambda: input().split()
MOD = 10 ** 9 + 7

def main(): 
    n, q = LII()
    nums = LII()
    mod = 10 ** 9 + 7
    rev2 = (mod + 1) // 2
    
    ans = [0] * (n + 1)
    
    cur = 0
    for v in reversed(nums):
        ans[v] = (ans[v] + cur * rev2) % mod
        cur = (cur + v) * rev2 % mod
    
    ans[1] = (ans[1] + cur) % mod
    
    print(*ans[1:], sep='\n')

if __name__ == "__main__":
    main()